components/measure-upload/MeasureRepositoryUpload.tsx

```diff
@@ -0,0 +1,235 @@
+    }
+    setMeasureBundle(mb => ({
+      ...mb,
+      measureRepositoryUrl: event.currentTarget.value,
+      selectedMeasureId: null,
+      displayMap: {},
+      ...(!mb.isFile ? { content: null } : undefined)
+    }));
+    setIsLoadingPackage(PACKAGE_STATES.NONE);
+  }
+
+  /*
+    Takes the entries from a Measure retrieval response and creates a map from Measure id
+    to Measure display label for Measure select component
+  */
+  function createMeasureDisplayMap(
+    displayMap: Record<string, string>,
+    { resource: measure }: { resource: fhir4.Measure }
+  ) {
+    let label: string;
+    const officialIdentifier = measure.identifier?.find(
+      identifier =>
+        identifier.use === 'official' && identifier.system === 'http://hl7.org/fhir/cqi/ecqm/Measure/Identifier/cms'
+    );
+    if (officialIdentifier?.value) {
+      label = measure.version ? `${officialIdentifier.value}|${measure.version}` : officialIdentifier.value;
+    } else if (measure.name) {
+      label = measure.version ? `${measure.name}|${measure.version}` : measure.name;
+    } else {
+      label = measure.id as string;
+    }
+    displayMap[measure.id as string] = label;
+    return displayMap;
+  }
+
+  /*
+    Sends a `GET` request to the `/Measure` endpoint of the specified Measure Repository Service.
+    Parses and stores the measureIds if successful
+  */
+  async function retrieveMeasures() {
+    setIsLoadingIds(true);
+    try {
+      const response = await fetch(`${measureRepositoryUrl}/Measure`);
+      const responseBody = await response.json();
+      if (response.status === 200) {
+        const displayMap = responseBody.entry.reduce(createMeasureDisplayMap, {});
+        setMeasureBundle(mb => ({ ...mb, displayMap }));
+        if (!measureSelectIsOpen) {
+          setMeasureSelectIsOpen(true);
+        }
+      } else {
+        showNotification({
+          icon: <IconAlertCircle />,
+          title: 'Failed reaching out to server',
+          message: `Server responded with code: ${response.status}:${response.statusText}`,
+          color: 'red'
+        });
+      }
+    } catch (e) {
+      showNotification({
+        icon: <IconAlertCircle />,
+        title: 'Failed reaching out to server',
+        message: `Could not connect to server with url: ${measureRepositoryUrl}`,
+        color: 'red'
+      });
+    }
+    setIsLoadingIds(false);
+  }
+
+  /*
+    Sends a `POST` request to the `Measure/:id/$package` endpoint of the specified Measure
+    Repository Service. Checks the resulting bundle is valid and saves it if so.
+  */
+  async function retrieveMeasurePackage(id: string) {
+    setMeasureBundle(mb => ({ ...mb, selectedMeasureId: id }));
+    if (id) {
+      const display = displayMap[id];
+      setIsLoadingPackage(PACKAGE_STATES.LOADING);
+      const response = await fetch(`${measureRepositoryUrl}/Measure/${id}/$package`, MEASURE_RETRIEVAL_OPTIONS);
+      const responseBody = await response.json();
```

Comment:
Looks like `responseBody` is showing up as an `any` type... I think we should amp up the typing in this function - since `responseBody` is of type `any`, everything else seems to be as well (like `measures`, etc.). Then we can avoid having to set `measures` as a `fhir4.BundleEntry[]` way later (I think)